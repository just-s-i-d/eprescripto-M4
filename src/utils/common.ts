import { notification } from "antd";

export const showToast = (message: string, type: "success" | "error") => {
  notification.open({
    message: type === "success" ? "Success" : "Error",
    description: message,
    type: type,
    className: `bg-${type}-bgColor border rounded border-${type}-borderColor`,
    duration: 3,
  });
};
