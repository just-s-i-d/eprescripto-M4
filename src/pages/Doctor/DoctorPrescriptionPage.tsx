import MedicationForm from "@components/Doctor/MedicationForm";
import NewPatientCard from "@components/Doctor/NewPatientCard";
import ErrorBoundary from "@components/ErrorBoundary";
import CardTitle from "@components/ui/CardTitle";
import PopModal from "@components/ui/PopModal";
import {
  ApiResponseData,
  PatientDataType,
  PatientInfoType,
  SelectOptionsType,
} from "@constants/types";
import {
  addNewPatient,
  createNewPrescription,
  getData,
  patientsEndpoint,
} from "@utils/Doctor";
import { showToast } from "@utils/common";
import { Button, Card, Form, Input, Select } from "antd";
import { useEffect, useState } from "react";
import { MedicationType } from "../../constants/types";
import {
  createOptions,
  medicationInputFieldGenerator,
} from "../../utils/Doctor";
import { SELECT_GENDER_OPTIONS } from "@constants/constants";
const { useForm, useWatch } = Form;
const { TextArea } = Input;
const DoctorPrescriptionPage = () => {
  const [open, setOpen] = useState(false);
  const [showAllergies, setShowAllergies] = useState(false);
  const [disabled, setDisabled] = useState(false);
  const [newPatientForm] = useForm();
  const [prescriptionForm] = useForm();
  const formValues = useWatch([], newPatientForm);
  const [patientsData, setPatientsData] = useState<PatientInfoType[]>();
  const [selectedPatient, setSelectedPatient] = useState<PatientInfoType>();
  const [options, setOptions] = useState<SelectOptionsType>();
  const prescriptionFormValues = useWatch([], prescriptionForm);
  const [medicationInputFields, setMedicationInputFiels] = useState<
    MedicationType[]
  >([medicationInputFieldGenerator()]);
  const addMedicationInput = () => {
    const inputField = medicationInputFieldGenerator();
    setMedicationInputFiels((prev) => [...prev, inputField]);
  };
  const removeMedicationInput = (id: number) => {
    const reducedInputFields = medicationInputFields.filter(
      (item) => item.id !== id,
    );
    setMedicationInputFiels(reducedInputFields);
  };
  const updatedData = () => {
    getData<ApiResponseData<PatientDataType>>(patientsEndpoint).then((res) => {
      const data = res.map((element) => ({
        id: element.id,
        ...element.attributes,
      }));
      setPatientsData(data);
      setOptions(createOptions(data));
    });
  };
  const onChangeHanlder = (id: number) => {
    if (patientsData) {
      const patient = patientsData.find(
        (patient: PatientInfoType) => patient.id === id,
      );
      setSelectedPatient(patient);
    }
  };
  const submitHanlder = () => {
    const patientInfo = formValues;
    addNewPatient(patientInfo)
      .then(() => {
        showToast("New patient added", "success");
        updatedData();
      })
      .catch((res) => {
        showToast(res.message, res.type);
      });
    newPatientForm.resetFields();
  };
  const cancelHanlder = () => {
    prescriptionForm.resetFields();
    setSelectedPatient(undefined);
  };
  const prescriptionFormSubmitHandler = () => {
    createNewPrescription(prescriptionFormValues)
      .then((res) => {
        showToast(res.message, res.type);
      })
      .catch((error) => {
        showToast(error.message, error.type);
      })
      .finally(() => {
        prescriptionForm.resetFields();
        setSelectedPatient(undefined);
      });
  };
  useEffect(() => {
    updatedData();
  }, []);
  useEffect(() => {
    prescriptionForm.resetFields();
  }, [selectedPatient]);
  useEffect(() => {
    prescriptionForm.validateFields({ validateOnly: true }).then(
      () => {
        setDisabled(false);
      },
      () => {
        setDisabled(true);
      },
    );
  }, [prescriptionFormValues]);

  return (
    <ErrorBoundary>
      <Card>
        <CardTitle className="flex justify-between">
          Prescription Form{" "}
          <Button
            className="text-white bg-secondary"
            onClick={() => setOpen(true)}
            data-testid="add-patient-btn"
          >
            New Patient
          </Button>
        </CardTitle>
        <Form
          data-testid="prescription-form"
          form={prescriptionForm}
          layout="vertical"
          className="custom-form custom-form-profile"
          initialValues={selectedPatient}
        >
          <div className="grid sm:grid-cols-2 md:grid-cols-3 xl:grid-cols-4 gap-x-6 gap-y-3">
            <Form.Item name="pId" label="Patient" className="w-full" required>
              <Select
                data-testid="patient-select"
                placeholder="Select a Patient"
                options={options}
                onChange={onChangeHanlder}
                value={selectedPatient?.id}
              />
            </Form.Item>
            <Form.Item
              className="w-full "
              name="pName"
              label="Patient Name"
              required
            >
              <Input disabled />
            </Form.Item>
            <Form.Item
              className="w-full"
              name="dob"
              label="Date of Birth"
              required
            >
              <Input type="date" disabled />
            </Form.Item>
            <Form.Item
              className="xs:w-full"
              name="gender"
              label="Gender"
              required
              rules={[
                {
                  required: true,
                  message: "Field cannot be empty",
                },
              ]}
            >
              <Select
                disabled
                placeholder="Select a gender"
                options={SELECT_GENDER_OPTIONS}
              />
            </Form.Item>
            <Form.Item
              name="prescriptionName"
              label="Prescription Name"
              className="xs:w-full"
              required
              rules={[
                {
                  required: true,
                  message: "Field cannot be empty",
                  validator: (_, value) => {
                    if (value?.trim()) return Promise.resolve();
                    else return Promise.reject();
                  },
                },
                {
                  pattern: /^[A-Za-z0-9\s-]+$/,
                  message: "Special character not allowed",
                },
                {
                  max: 20,
                  message: "Only 20 characters allowed",
                },
              ]}
            >
              <Input />
            </Form.Item>
            <Form.Item
              className="xs:w-full"
              name="nextVisit"
              label="Next Visit"
              rules={[{ pattern: /^[0-9]+$/, message: "Enter valid integer" }]}
            >
              <Input type="number" placeholder="In Days" />
            </Form.Item>
            <Form.Item name="notes" label="Important Notes" className="w-full">
              <TextArea rows={1} />
            </Form.Item>
          </div>
          <div className="w-full mt-4">
            {medicationInputFields.map((item, index) => (
              <MedicationForm
                key={item.id}
                fields={item}
                removeMedicationInput={removeMedicationInput}
                removeable={index > 0}
              />
            ))}
            <div className="flex justify-between mt-2 mb-8">
              <Button onClick={addMedicationInput}>Add medication</Button>
              <Button onClick={() => setShowAllergies(true)}>
                Show Patient Allergies
              </Button>
            </div>
          </div>
          <div className="flex gap-4">
            <Button
              onClick={prescriptionFormSubmitHandler}
              disabled={disabled}
              className="text-white bg-secondary disabled:opacity-50"
            >
              Create Prescription
            </Button>
            <Button onClick={cancelHanlder}>Cancel</Button>
          </div>
        </Form>
      </Card>
      <ErrorBoundary>
        <PopModal
          open={open}
          setOpen={setOpen}
          title="New Patient"
          okButtonText="Add Patient"
          confirmHandler={submitHanlder}
          footer={true}
        >
          <NewPatientCard newPatientForm={newPatientForm} />
        </PopModal>
      </ErrorBoundary>
      <ErrorBoundary>
        <PopModal
          footer={false}
          open={showAllergies}
          setOpen={setShowAllergies}
          title="Patient Allergies"
        >
          <div className="min-h-[10vh] max-h-[20vh] mt-4 text-lg overflow-scroll">
            {selectedPatient?.allergies ? (
              <>
                <div>Patient has following allergies :</div>
                <div>{selectedPatient?.allergies}</div>
              </>
            ) : (
              <div>Patient has no allergies mentioned</div>
            )}
          </div>
        </PopModal>
      </ErrorBoundary>
    </ErrorBoundary>
  );
};
export default DoctorPrescriptionPage;
