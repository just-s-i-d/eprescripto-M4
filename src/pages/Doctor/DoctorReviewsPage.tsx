import { useEffect } from "react";

import { Card, Table } from "antd";
import { ColumnsType } from "antd/es/table";

import { ApiResponseData, ReviewDataType } from "@constants/types";
import { getData, reviewsDataEndPoint } from "@utils/Doctor";
import useStatesHook from "../../hooks/useStatesHook";
import starPng from "@assets/star.png";
import ErrorBoundary from "@components/ErrorBoundary";

const DoctorReviewsPage = () => {
  const reviews = useStatesHook<Record<string, string | number>[]>();
  const columns: ColumnsType<ReviewDataType> = [
    {
      title: "S. No",
      render: (_, _1, index) => index + 1,
    },
    {
      title: "Rating",
      dataIndex: "rating",
      key: "rating",
      render: (rating: number) => (
        <span className="flex">
          {Array.from({ length: rating }, (_, index) => (
            <img
              src={starPng}
              key={index}
              role="img"
              aria-label="star"
              width={20}
            />
          ))}
        </span>
      ),
      filters: [
        {
          text: "1 Star",
          value: 1,
        },
        {
          text: "2 Star",
          value: 2,
        },
        {
          text: "3 Star",
          value: 3,
        },
        {
          text: "4 Star",
          value: 4,
        },
        {
          text: "5 Star",
          value: 5,
        },
      ],
      onFilter: (value, record) => record.rating === value,
      sorter: {
        compare: (a, b) => a.rating - b.rating,
        multiple: 3,
      },
    },
    {
      title: "Comment",
      dataIndex: "comment",
      key: "comment",
    },
  ];
  useEffect(() => {
    getData<ApiResponseData<Record<string, string | number>>>(
      reviewsDataEndPoint,
    )
      .then((res) => {
        const data = res.map((element) => element.attributes);
        reviews.setData(data);
        reviews.setError(false);
        reviews.setLoading(false);
      })
      .catch(() => {
        reviews.setError(true);
        reviews.setLoading(false);
      });
  }, [reviews.refresh]);
  return (
    <>
      <Card className="overflow-scroll min-h-[50vh]">
        <ErrorBoundary
          error={reviews.error}
          refreshComponent={() => reviews.setRefresh((prev) => !prev)}
        >
          <Table
            className="custom-table min-w-[800px]"
            columns={columns as ColumnsType<Record<string, string | number>>}
            dataSource={reviews.data}
            loading={reviews.loading}
          />
        </ErrorBoundary>
      </Card>
    </>
  );
};
export default DoctorReviewsPage;
