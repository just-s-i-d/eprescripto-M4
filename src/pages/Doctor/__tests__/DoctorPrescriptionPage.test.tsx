import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@test/matchMedia";
import DoctorPrescriptionPage from "../DoctorPrescriptionPage";
import { act } from "react-dom/test-utils";
import { addNewPatient, createNewPrescription } from "@utils/Doctor";
import { showToast } from "@utils/common";

jest.mock("@utils/Doctor", () => {
  return {
    ...jest.requireActual("@utils/Doctor"),
    getData: jest.fn().mockResolvedValue([]),
    addNewPatient: jest.fn(),
    createNewPrescription: jest.fn(),
  };
});

jest.mock("@utils/common", () => ({
  showToast: jest.fn(),
}));

export const mockedPrescriptionFormData = {
  dob: "2001-05-27",
  dosage2: "500mg",
  gender: "male",
  instruction2: "beforeFood",
  medicine2: "diclovin",
  pName: "siddharth paneri",
  prescriptionName: "record 2",
  timesPerDay2: 2,
};
describe("Doctor Prescriptions Page tests", () => {
  it("Renders the prescription form", () => {
    const { asFragment } = render(<DoctorPrescriptionPage />);
    expect(screen.getByText("Prescription Form")).toBeInTheDocument();
    expect(screen.getByTitle("Patient")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("Prescription Form cancel test", () => {
    render(<DoctorPrescriptionPage />);
    const cancelBtn = screen.getByText("Cancel");
    expect(cancelBtn).toBeInTheDocument();
    fireEvent.click(cancelBtn);
  });

  it("Add new patient modal test", async () => {
    (addNewPatient as jest.Mock).mockResolvedValue("Mocked Result");
    render(<DoctorPrescriptionPage />);
    const addNewPatientBtn = screen.getByRole("button", {
      name: "New Patient",
    });
    expect(addNewPatientBtn).toBeInTheDocument();
    act(() => {
      fireEvent.click(addNewPatientBtn);
    });
    await waitFor(() => {
      expect(
        screen.getByRole("dialog", { name: "New Patient" }),
      ).toBeInTheDocument();
      const addBtn = screen.getByText("Add Patient");
      fireEvent.click(addBtn);
    });
    await waitFor(() => {
      expect(addNewPatient).toHaveBeenCalled();
      expect(showToast).toHaveBeenCalled();
    });
  });

  it("Show patient allergies modal test", async () => {
    render(<DoctorPrescriptionPage />);
    const showPatientAllergiesBtn = screen.getByRole("button", {
      name: "Show Patient Allergies",
    });
    expect(showPatientAllergiesBtn).toBeInTheDocument();
    act(() => {
      fireEvent.click(showPatientAllergiesBtn);
    });
    await waitFor(() => {
      expect(screen.getByText("Patient Allergies")).toBeInTheDocument();
    });
  });

  it("Prescription Form submit test successfull", async () => {
    (createNewPrescription as jest.Mock).mockResolvedValue({
      message: "Created",
      type: "success",
    });
    render(<DoctorPrescriptionPage />);
    const prescriptionForm = screen.getByTestId("prescription-form");
    expect(prescriptionForm).toBeInTheDocument();
    expect(screen.getByLabelText("Prescription Name")).toBeInTheDocument();
    fireEvent.change(screen.getByLabelText("Patient"), {
      target: { value: 1 },
    });
    fireEvent.change(screen.getByLabelText("Patient Name"), {
      target: { value: mockedPrescriptionFormData.pName },
    });
    fireEvent.change(screen.getByLabelText("Date of Birth"), {
      target: { value: mockedPrescriptionFormData.dob },
    });
    fireEvent.change(screen.getByLabelText("Gender"), {
      target: { value: mockedPrescriptionFormData.gender },
    });
    fireEvent.change(screen.getByLabelText("Prescription Name"), {
      target: { value: mockedPrescriptionFormData.prescriptionName },
    });
    fireEvent.change(screen.getByLabelText("Medicine Name"), {
      target: { value: mockedPrescriptionFormData.medicine2 },
    });
    fireEvent.change(screen.getByLabelText("Dosage"), {
      target: { value: mockedPrescriptionFormData.dosage2 },
    });
    fireEvent.change(screen.getByLabelText("Frequency"), {
      target: { value: mockedPrescriptionFormData.timesPerDay2 },
    });
    fireEvent.change(screen.getByLabelText("Instruction"), {
      target: { value: mockedPrescriptionFormData.instruction2 },
    });
    fireEvent.submit(prescriptionForm);
    await waitFor(() => {
      const submitFormBtn = screen.getByRole("button", {
        name: "Create Prescription",
      });
      fireEvent.click(submitFormBtn);
      expect(createNewPrescription).toHaveBeenCalled();
      expect(showToast).toHaveBeenCalledWith("Created", "success");
    });
  });
});
