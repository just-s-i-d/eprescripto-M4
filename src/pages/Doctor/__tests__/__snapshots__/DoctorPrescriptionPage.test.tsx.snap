// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor Prescriptions Page tests Renders the prescription form 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-card-body"
    >
      <h1
        class="text-xl font-bold mb-6 flex justify-between"
      >
        Prescription Form 
        <button
          class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default text-white bg-secondary"
          data-testid="add-patient-btn"
          type="button"
        >
          <span>
            New Patient
          </span>
        </button>
      </h1>
      <form
        class="ant-form ant-form-vertical css-dev-only-do-not-override-1adbn6x custom-form custom-form-profile"
        data-testid="prescription-form"
      >
        <div
          class="grid sm:grid-cols-2 md:grid-cols-3 xl:grid-cols-4 gap-x-6 gap-y-3"
        >
          <div
            class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="pId"
                  title="Patient"
                >
                  Patient
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      aria-required="true"
                      class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow"
                      data-testid="patient-select"
                    >
                      <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-controls="pId_list"
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-owns="pId_list"
                            aria-required="true"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            id="pId"
                            readonly=""
                            role="combobox"
                            style="opacity: 0;"
                            type="search"
                            unselectable="on"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-placeholder"
                        >
                          Select a Patient
                        </span>
                      </div>
                      <span
                        aria-hidden="true"
                        class="ant-select-arrow"
                        style="user-select: none;"
                        unselectable="on"
                      >
                        <span
                          aria-label="down"
                          class="anticon anticon-down ant-select-suffix"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item w-full  css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="pName"
                  title="Patient Name"
                >
                  Patient Name
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                      disabled=""
                      id="pName"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="dob"
                  title="Date of Birth"
                >
                  Date of Birth
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                      disabled=""
                      id="dob"
                      type="date"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item xs:w-full css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="gender"
                  title="Gender"
                >
                  Gender
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      aria-required="true"
                      class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow ant-select-disabled"
                    >
                      <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-controls="gender_list"
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-owns="gender_list"
                            aria-required="true"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            disabled=""
                            id="gender"
                            readonly=""
                            role="combobox"
                            style="opacity: 0;"
                            type="search"
                            unselectable="on"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-placeholder"
                        >
                          Select a gender
                        </span>
                      </div>
                      <span
                        aria-hidden="true"
                        class="ant-select-arrow"
                        style="user-select: none;"
                        unselectable="on"
                      >
                        <span
                          aria-label="down"
                          class="anticon anticon-down ant-select-suffix"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item xs:w-full css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class="ant-form-item-required"
                  for="prescriptionName"
                  title="Prescription Name"
                >
                  Prescription Name
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input css-dev-only-do-not-override-1adbn6x"
                      id="prescriptionName"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item xs:w-full css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class=""
                  for="nextVisit"
                  title="Next Visit"
                >
                  Next Visit
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input css-dev-only-do-not-override-1adbn6x"
                      id="nextVisit"
                      placeholder="In Days"
                      type="number"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
              >
                <label
                  class=""
                  for="notes"
                  title="Important Notes"
                >
                  Important Notes
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <textarea
                      class="ant-input css-dev-only-do-not-override-1adbn6x"
                      id="notes"
                      rows="1"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full mt-4"
        >
          <div
            class="mt-4 mb-2 flex justify-between"
          >
            <h2
              class="text-xl"
            >
              Medication
            </h2>
          </div>
          <div
            class="w-full grid sm:grid-cols-2 md:grid-cols-4 gap-x-5 gap-y-3"
          >
            <div
              class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                >
                  <label
                    class="ant-form-item-required"
                    for="medicine1"
                    title="Medicine Name"
                  >
                    Medicine Name
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-1adbn6x"
                        id="medicine1"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                >
                  <label
                    class="ant-form-item-required"
                    for="dosage1"
                    title="Dosage"
                  >
                    Dosage
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-1adbn6x"
                        id="dosage1"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                >
                  <label
                    class="ant-form-item-required"
                    for="timesPerDay1"
                    title="Frequency"
                  >
                    Frequency
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        aria-required="true"
                        class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-controls="timesPerDay1_list"
                              aria-expanded="false"
                              aria-haspopup="listbox"
                              aria-owns="timesPerDay1_list"
                              aria-required="true"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="timesPerDay1"
                              readonly=""
                              role="combobox"
                              style="opacity: 0;"
                              type="search"
                              unselectable="on"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-placeholder"
                          />
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex gap-6"
            >
              <div
                class="ant-form-item w-full css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="instruction1"
                      title="Instruction"
                    >
                      Instruction
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          aria-required="true"
                          class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-autocomplete="list"
                                aria-controls="instruction1_list"
                                aria-expanded="false"
                                aria-haspopup="listbox"
                                aria-owns="instruction1_list"
                                aria-required="true"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="instruction1"
                                readonly=""
                                role="combobox"
                                style="opacity: 0;"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-placeholder"
                            />
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select: none;"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex justify-between mt-2 mb-8"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default"
              type="button"
            >
              <span>
                Add medication
              </span>
            </button>
            <button
              class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default"
              type="button"
            >
              <span>
                Show Patient Allergies
              </span>
            </button>
          </div>
        </div>
        <div
          class="flex gap-4"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default text-white bg-secondary disabled:opacity-50"
            type="button"
          >
            <span>
              Create Prescription
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default"
            type="button"
          >
            <span>
              Cancel
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
